### DISCUSSION TITLE
R - gagal melakukan full join. cannot allocate vector of size ...

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#134] [Q&A] [answered]
![Capture](https://user-images.githubusercontent.com/95233522/173790343-c5d81c24-10d4-4883-85cd-61b2d9b6eb67.JPG)
So, this happened when I tried to merge several tables into one. Does this size limit can be safely changed for a while? 

Mau coba atasi dengan memory. limit() tapi takut bikin rusak. Apakah aman?

**Comment by inytss** [comment, marked as answer]
Dear @Hertzspr ,
Saya ingin mengkonfirmasi terlebih dahulu untuk ukuran pada masing-masing object yang dilakukan full_join apakah memiliki jumlah row dan kolom yang cukup banyak? Apakah jumlah row nya sama antar object data frame tersebut? 
Apabila dilihat dari error yang ada, sepertinya memang ukuran data terlalu besar sehingga proses full_join tidak berhasil. Untuk mengatasinya Bapak dapat menggunakan `memory.limit()`. Untuk kerusakan object `merged_top5` tidak akan mempengaruhi pada object-object lain yang sudah Bapak buat sebelumnya.

**Reply by Hertzspr** [reply]
![obj](https://user-images.githubusercontent.com/95233522/173979982-932fc358-2cef-42a7-a816-175d86c7af8f.JPG)
Sepertinya tidak terlalu besar. Penggunaan memory.limitnya bagaimana ya, kemarin saya coba gagal.

**Reply by inytss** [reply]
Bisa menggunakan code berikut
```
memory.limit(size=56000)
```
Saran saya jika isi dari dataframe yang akan dimasukkan sama, dan yang membedakan adalah jumlah kolomnya, Bapak bisa menggunakan `cbind.data.frame()` atau dapat menggunakan `left_join()`. Selain itu jika pada object tersebut banyak kolom yang sama, Bapak dapat menggunakan id beberapa kolom yang sama tersebut agar tidak menambahkan kolom baru dengan nilai yang sama. Contoh, terdapat kolom `Player`, `Name_Player` dan `Rating` yang sama. Bapak dapat menggunakan code berikut dapat melakukan join.
```
df %>%
 full_join(df2) %>%
 full_join(df3)
```

**Reply by Hertzspr** [reply]
Saya belum paham maksud tulisan di paragraf dan contoh codenya.

**Reply by Hertzspr** [reply]
![setelah memory limit 56000](https://user-images.githubusercontent.com/95233522/174246529-62e83d0a-0c91-461d-bf1e-77c4fe96ab2b.JPG)

Eh tapi setelah memory.limit(size=56000) ini berhasil ke merge. Terima kasih. Cuma mau tanya, penasaran itu 56000 maksudnya 56000 MB ya? Apa gimana?

**Reply by inytss** [reply]
Betul pak, 56000 disitu adalah 56000MB.
